### PR TITLE
:bug: Ensure that env variable PODMAN_BIN wins

### LINKED ISSUE #107: [BUG] Kantra must obey user provided PODMAN_BIN environment variable
As a user of kantra I would like to use Docker instead of Podman by setting the environment variable `PODMAN_BIN` to `$(which docker)`.
Unfortunately, if there is a `podman` executable on `${PATH}`, [the environment variable is ignored and Podman is used instead](https://github.com/ascheman/kantra/blob/b7180bb69033d4f3d6a4efce201da042ae2f8305/cmd/settings.go#L29).

This leads to errors like this:
```
ERRO[0000] container run error                           error="exit status 125"
ERRO[0000] failed listing labels                         error="Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM\nError: unable to connect to Podman socket: failed to connect: dial tcp 127.0.0.1:49680: connect: connection refused\n"
ERRO[0000] failed to list rule labels                    error="Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM\nError: unable to connect to Podman socket: failed to connect: dial tcp 127.0.0.1:49680: connect: connection refused\n"
Error: Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM
```

### PR DESCRIPTION
**Pull request by ascheman**
Fixes #107 
As discussed [via Slack](https://kubernetes.slack.com/archives/C04QZJFQ0UA/p1698853574068369?thread_ts=1698830510.500689&cid=C04QZJFQ0UA)

### CODE REVIEW

File: cmd/settings.go [hunk 1]
```diff
@@ -26,10 +26,14 @@ type Config struct {
 }
 
 func (c *Config) Load() error {
-	podmanPath, _ := exec.LookPath("podman")
-	if podmanPath != c.PodmanBinary && (podmanPath != "" || len(podmanPath) > 0) {
-		os.Setenv("PODMAN_BIN", podmanPath)
+	var envValue = os.Getenv("PODMAN_BIN")
```
Comment (pranavgaikwad) [inline cmd/settings.go:29]: you might know this already...idiomatic way of initiating a variable with value is: 

```
envValue := os.Getenv("PODMAN_BIN")
```

**Review by pranavgaikwad — APPROVED** [review]
LGTM

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
Requires signing off

**Review by eemcmullan — APPROVED** [review]

File: cmd/settings.go [hunk 2]
```diff
@@ -26,10 +26,14 @@ type Config struct {
...
+	if envValue == "" {
+		podmanPath, _ := exec.LookPath("podman")
```
Comment (shawn-hurley) [inline cmd/settings.go:31]: Worried about eating the error here

It does seem that LookPath has some interesting edge case:

> In older versions of Go, LookPath could return a path relative to the current directory. As of Go 1.19, LookPath will instead return that path along with an error satisfying errors.Is(err, ErrDot). See the package documentation for more details.

Can you also add a check if err is not nil && err is not ErrDot? if it is error I think this means that Rodman is not installed and we can error out noting that. @pranavgaikwad and @eemcmullan will defer to y'all if I understood that correctly :)

Comment (eemcmullan) [reply cmd/settings.go:31]: I think this sounds smart. I don't remember the original intention behind masking the error here, but +1 on checking for those two.

Comment (ascheman) [reply cmd/settings.go:31]: I would like to concentrate on correct handling of the PODMAN_BIN variable here, as the bug #107 and the PR #108 is about that problem. If you want to check whether `podman` is installed at all and abort Kantra when this is not the case, this is a different concern. OK?

Comment (shawn-hurley) [reply cmd/settings.go:31]: Considering that this was prior behavior, I think it is reasonable that we can do it in a follow up, sorry just caught it now, but that makes sense. 

Created a follow up issue: https://github.com/konveyor/kantra/issues/113

**Review by shawn-hurley — APPROVED** [review]

**Review by pranavgaikwad — APPROVED** [review]

**Comment by ascheman** [comment]
Thanks!

**Pull request merged**